database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -85,11 +84,26 @@ public String parseSnapshot(DataSnapshot snapshot) {
                 return snapshot.getKey();
             }
         });
+    }
 
+    @Override
+    protected void onCreate() {
+        super.onCreate();
         mKeySnapshots.addChangeEventListener(this);
     }
 
     @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        mKeySnapshots.removeChangeEventListener(this);
+
+        for (DatabaseReference ref : mRefs.keySet()) {
+            ref.removeEventListener(mRefs.get(ref));
+        }
+        mRefs.clear();
```

Comment:
In the previous version we called `clearData()` but now we only call `mRefs.clear()`, is this intentional?